apps/remix-ide/src/app/tabs/runTab/settings.js

```diff
@@ -390,29 +390,32 @@ class SettingsUI {
   }
 
   // TODO: unclear what's the goal of accountListCallId, feels like it can be simplified
-  fillAccountsList () {
+  async fillAccountsList () {
     this.accountListCallId++
-    var callid = this.accountListCallId
-    var txOrigin = this.el.querySelector('#txorigin')
-    this.blockchain.getAccounts((err, accounts) => {
+    const callid = this.accountListCallId
+    const txOrigin = this.el.querySelector('#txorigin')
+    try {
+      let accounts = await this.blockchain.getAccounts()
```

Comment:
if this function throws, migtht be that we still want the accounts list to be cleared in the UI.